rustls/src/error.rs

```diff
@@ -372,7 +375,7 @@ impl From<CertificateError> for AlertDescription {
             // certificate_unknown
             //  Some other (unspecified) issue arose in processing the
             //  certificate, rendering it unacceptable.
-            Other(_) => Self::CertificateUnknown,
+            Other(_) | UnknownRevocationStatus => Self::CertificateUnknown,
```

Comment:
I'm not sure this mapping from the new unknown revocation status certificate error to alert makes sense. Suggestions welcome.

Reply:
Just in terms of names, `BadCertificate` seems more appropriate than `CertificateUnknown`? Did you check what BoringSSL/OpenSSL do?

Reply:
>  Did you check what BoringSSL/OpenSSL do?

I find the BoringSSL/OpenSSL code pretty hard to follow but I'll put some legwork in and see what I can figure out. 

> Just in terms of names, BadCertificate seems more appropriate than CertificateUnknown?

Switched to `BadCertificate` in the meantime.

Reply:
> I'll put some legwork in and see what I can figure out

I believe for both OpenSSL and BoringSSL in the absence of callbacks doing something differently, the failure to find a relevant CRL for a certificate in-scope for revocation checking will be turned into a `X509_V_ERR_UNABLE_TO_GET_CRL` result ([OpenSSL](https://github.com/openssl/openssl/blob/fac61ea4618c83826b51aebf03cbc2bc3ac7b8c8/crypto/x509/x509_vfy.c#L1015-L1024), [BoringSSL](https://github.com/google/boringssl/blob/bfa8369795b7533a222a72b7a1bc928941cd66bf/crypto/x509/x509_vfy.c#L832-L843)). In both cases that error is mapped to an Unknown CA alert ([OpenSSL](https://github.com/openssl/openssl/blob/fac61ea4618c83826b51aebf03cbc2bc3ac7b8c8/ssl/statem/statem_lib.c#L1763), [BoringSSL](https://github.com/google/boringssl/blob/bfa8369795b7533a222a72b7a1bc928941cd66bf/ssl/ssl_x509.cc#L1330C1-L1330C1)).

We could do the same, but at first pass it feels to me like `BadCertificate` might be a better fit? I haven't been able to find any standards text that would suggest which is more appropriate.

Reply:
> We could do the same, but at first pass it feels to me like BadCertificate might be a better fit? I haven't been able to find any standards text that would suggest which is more appropriate.

Since the existing `From<CertificateError> for AlertDescription ` impl is commented as "heavily referenced" from the OpenSSL/BoringSSL implementations I decided to tack on a commit that updates our handling of `Error::UnknownRevocationStatus` to use `AlertDescription::UnknownCA` for this condition too: 155fbd6b0f4c51eb0f7b249c0ea90dba5d1d3ad4

It's probably the least surprising thing to do and I can't find strong justification for doing otherwise :shrug: We can always change it again if someone disagrees with my logic here.